blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractRunImpl.java

```diff
@@ -326,7 +326,21 @@ private boolean isCompletedOrAborted(){
 
     @Override
     public Links getLinks() {
-        return super.getLinks().add("parent", parent.getLink());
+        Links links = super.getLinks().add("parent", parent.getLink());
+        Run nextRun = run.getNextBuild();
+        Run prevRun = run.getPreviousBuild();
+
+        if(nextRun != null) {
+            Link nextRunLink = organization.getLink().rel(String.format("pipelines/%s/runs/%s", run.getParent().getName(), nextRun.getId()));
```

Comment:
Use `LinkResolver.resolveLink(nextRun)` instead of hand crafting run link, I suspect it might not work when inside nested folders.